File: Engine/src/main/java/com/render/engine/util/LogUtil.java
package com.render.engine.util;

import android.util.Log;

/**
 * @author lyzirving
 */
public class LogUtil {
    private static final String TAG = "Engine";

    public static void d(String tag, String msg) {
        Log.d(buildTag(tag), msg);
    }

    public static void i(String tag, String msg) { Log.i(buildTag(tag), msg); }

    public static void e(String tag, String msg) {
        Log.e(buildTag(tag), msg);
    }

    private static String buildTag(String tag) {
        return TAG + "_" + tag;
    }
}

File: Engine/src/main/cpp/impl/common/LogUtil.cpp
#include <android/log.h>
#include "LogUtil.h"

#define TAG "Engine_Native"

void LogUtil::logD(std::string tag, std::initializer_list<std::string> msg) {
    std::string result = std::move(tag);
    result.append("_");
    for (const auto & tmp : msg) {
        result.append(tmp);
    }
    __android_log_print(ANDROID_LOG_DEBUG, TAG, "%s", result.c_str());
}

void LogUtil::logI(std::string tag, std::initializer_list<std::string> msg) {
    std::string result = std::move(tag);
    result.append("_");
    for (const auto & tmp : msg) {
        result.append(tmp);
    }
    __android_log_print(ANDROID_LOG_INFO, TAG, "%s", result.c_str());
}

void LogUtil::logE(std::string tag, std::initializer_list<std::string> msg) {
    std::string result = std::move(tag);
    result.append("_");
    for (const auto & tmp : msg) {
        result.append(tmp);
    }
    __android_log_print(ANDROID_LOG_ERROR, TAG, "%s", result.c_str());
}

